### DISCUSSION TITLE
2023/09/29/enum-in-rust

### DISCUSSION THREAD

**Discussion by giscus** [kaochenlong/kaochenlong-discuss#9] [General]
# 2023/09/29/enum-in-rust

在寫程式的時候，雖然對電腦來說都是 0 跟 1，但對身為開發者的人類來說有好的命名或識別是很重要的，對自己好，對跟你一起工作的夥伴也好。這個章節要介紹的「列舉（Enum）」並不是什麼很新或很特別的設計，在其它程式語言也常見，列舉是用來表示某種特定類型的值集合，通常會要把同樣類型的東西放在一起（例如顏色 Color...

https://kaochenlong.com/2023/09/29/enum-in-rust.html

**Comment by Ethkuil** [comment]
本质上的区别：Struct是积类型，Enum是和类型。

话说博主的「为你自己学」系列真是写得通俗易懂，读起来酣畅淋漓。

**Comment by afunTW** [comment]
> Struct 裡面不能包 Enum

請問一下這是什麼意思？我嘗試 `cargo run` 以下範例是正確的

```rust
enum Status {
    Active,
    Inactive,
    Pending,
}

struct User {
    name: String,
    age: u32,
    status: Status,
}

fn main() {
    let user = User {
        name: String::from("Alice"),
        age: 30,
        status: Status::Active,
    };

    match user.status {
        Status::Active => println!("User is active"),
        Status::Inactive => println!("User is inactive"),
        Status::Pending => println!("User is pending approval"),
    }
}
```

output: `User is active`

**Reply by kaochenlong** [reply]
我記得的確是可以包沒錯，但我忘了當時為什麼我會這樣寫，我先把這行移除，感謝提醒 :)
